React/Views/RCTRefreshControl.m

```diff
@@ -132,9 +144,15 @@ - (void)setRefreshing:(BOOL)refreshing
   }
 }
 
+- (void)setCurrentRefreshingState:(BOOL)refreshing
+{
+  _currentRefreshingState = refreshing;
+  _currentRefreshingStateTimestamp = _currentRefreshingStateClock++;
```

Comment:
currentRefreshingStateClock is never decremented. Was this meant to track/prevent other potential races where it could go above 1 or 2?

Reply:
@matthargett no, it shouldn't because I used logical clock. Every time `refreshSetCurrentRefreshingState` is called i need unique timestamp. This time stamp is used in animation window which starts on 
https://github.com/facebook/react-native/pull/21763/files#diff-cfafa8d7e42ab87ab863f917e187a4aeR56
and when animation window ends 
https://github.com/facebook/react-native/pull/21763/files#diff-cfafa8d7e42ab87ab863f917e187a4aeR70
and timestamp is still the same, then state can be safely set, otherwise it should be thrown away due there is another animation in progress with higher clock timestamp.

Reply:
I've just changed that due i used local date time before, and as @cpojer has noted changing os date time can have impact on that logic. But in real life it wouldn't harm logic at all because chance that somebody changes os date time, and in that moment will trigger race condition described above, is astronomically low. But to prevent confusion, I've rid of that logic and replaced local date time with logical clock.